homeassistant/components/websocket_api/commands.py

```diff
@@ -778,7 +778,32 @@ async def handle_execute_script(
 
     context = connection.context(msg)
     script_obj = Script(hass, script_config, f"{const.DOMAIN} script", const.DOMAIN)
-    script_result = await script_obj.async_run(msg.get("variables"), context=context)
+    try:
+        script_result = await script_obj.async_run(
+            msg.get("variables"), context=context
+        )
+    except ServiceValidationError as err:
```

Comment:
Do more WebSocket commands raise a `ServiceValidationError`?

Reply:
I guess there are more services that are going to raise `ServiceValidationError` if they are caused by a user error.

Reply:
All websocket commands don't call services. I don't think we should handle ServiceValidationError in the common exception handler.

Reply:
They do in case a a script is called, as we do in the service panel of the developer tools.

Reply:
I'm not following. All websocket commands in Home Assistant are not related to services.

Reply:
When a service is executed from the developer service panel, it is wrapped into a script. So when e.g. an invalid preset mode is given, this should throw a `ServiceValidationError`. I've been testing the whole chain to be sure it works that way.
Or course it would work to if we would only process `HomeAssistantError` as `ServiceValidationError` is a subclass.

Reply:
Right. What I'm saying is that websocket commands can do much more than calling services. In the common exception handler `async_handle_exception` we should only handle exceptions that are common for all websocket commands. Exceptions that are raised specifically from service handlers should be handled specifically in the websocket commands that call those services.

Reply:
So what you are saying is that I could partly reimplement my first commit where I added `ServiceValidationError` to the websockets as websocket commands handler